kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/CsrServiceImpl.java

```diff
@@ -320,6 +326,29 @@ private CallSummaryRecord aggregateSummaryRecord(CallSummaryRecordDto csr) {
     }
 
 
+    private void validateCallSummaryRecord(Subscription subscription, CallSummaryRecordDto csr) throws InvalidCdrData {
+        if (!subscription.getSubscriptionPack().hasMessageWithWeekId(csr.getWeekId())) {
+            throw new InvalidCdrData(String.format("%s is an invalid weekId for pack %s",
+                    csr.getWeekId(), subscription.getSubscriptionPack().getName()));
+        }
+
+        if (!subscription.getSubscriptionPack().hasMessageWithFilename(csr.getContentFileName())) {
+            throw new InvalidCdrData(String.format("%s is an invalid contentFilename for pack %s",
+                    csr.getContentFileName(), subscription.getSubscriptionPack().getName()));
+        }
+
+        if (circleService.getByName(csr.getCircle()) == null) {
```

Comment:
Hmm, given the essence of this check-in was to make things faster, Should I remove these checks altogether? It seems to be much faster now, so I'm thinking of leaving them, but I guess they should be prime chop candidates.